src/plugins/data_source/server/routes/data_source_connection_validator.ts

```diff
@@ -14,10 +14,21 @@ export class DataSourceConnectionValidator {
 
   async validate() {
     try {
+      let validationResponse;
       // Amazon OpenSearch Serverless does not support .info() API
-      if (this.dataSourceAttr.auth?.credentials?.service === SigV4ServiceName.OpenSearchServerless)
-        return await this.callDataCluster.cat.indices();
-      return await this.callDataCluster.info();
+      if (
+        this.dataSourceAttr.auth?.credentials?.service === SigV4ServiceName.OpenSearchServerless
+      ) {
+        validationResponse = await this.callDataCluster.cat.indices();
+      } else {
+        validationResponse = await this.callDataCluster.info();
+      }
+
+      if (validationResponse?.statusCode === 200) {
```

Comment:
Can we add one more check here if have clustername or version for opensearch, and body if empty for serverless

Reply:
That's a good idea. I will update the validation logic into following conditions:

1. Server cluster:
test connection pass when: (response.statuscode == 200 && response.body.cluster_name.isNotEmpty())

2. Serverless collection:
test connection pass when: (response.statuscode == 200 && response.body.isNotEmpty())

cc @Flyingliuhub @bandinib-amzn